### PR TITLE
Delegates classes and keywords. Fixes #238

### LINKED ISSUE #238: Delegates: article on the Delegate class and delegate keyword
This is a sub-task for #63.

Write the article for the delegate class and keyword.

### PR DESCRIPTION
**Pull request by BillWagner**
This relies on PR #246

### CODE REVIEW

File: docs/languages/csharp/delegate-class.md [hunk 1]
```diff
@@ -1,11 +1,213 @@
...
+The compiler generates a class, derived from `System.Delegate`
+that matches the signature used (in this case, a method that
+returns an integer, and has two arguments). The type
+of that delegate is `Comparison`. The `Comparison` delegate
+type is a generic type. For details on generics see [here](generics.md).
+
+Notice that the syntax may appear as though it is declaring
+a variable, but it is actually declaring a *type*. You can
+define delegate types inside classes, directly inside namespaces,
+or evern in the global namespace.
```
Comment (ghogen) [inline docs/languages/csharp/delegate-class.md]: evern -> even

File: docs/languages/csharp/delegate-class.md [hunk 2]
```diff
@@ -1,11 +1,213 @@
...
+Using Lambda Expressions for delegate targets
+is covered more in a [later section](delegates-patterns.md).
+
+The Sort() example typically attaches a single target method to the
+delegate. However, delegate objects do support invocation lists that
+have multiple target methods attached to a delegate object.
+
+## Delegate and MulticastDelegate classes
+
+The language support decscribed above provides the features
```
Comment (ghogen) [inline docs/languages/csharp/delegate-class.md]: decscribed : spelling

**Comment by ghogen** [comment]
LGTM with a few minor comments

**Comment by BillWagner** [comment]
Updated per @ghogen comments, rebased and squashed.

**Comment by ghogen** [comment]
@blackdwarf @cartermp do you want to also review or shall we merge?

**Comment by cartermp** [comment]
Took a look and I think it's good.  Let's merge this puppy.

**Pull request merged**